cpp/src/utilities/cython.cu

```diff
@@ -789,6 +790,65 @@ std::unique_ptr<cy_multi_edgelists_t> call_egonet(raft::handle_t const& handle,
     CUGRAPH_FAIL("vertexType/edgeType combination unsupported");
   }
 }
+// Wrapper for graph generate_rmat_edgelist()
+// to expose the API to cython
+// enum class generator_distribution_t { POWER_LAW = 0, UNIFORM };
+template <typename vertex_t>
+std::unique_ptr<graph_generator_t> call_generate_rmat_edgelist(raft::handle_t const& handle,
+                                                               size_t scale,
+                                                               size_t num_edges,
+                                                               double a,
+                                                               double b,
+                                                               double c,
+                                                               uint64_t seed,
+                                                               bool clip_and_flip,
+                                                               bool scramble_vertex_ids)
+{
+  auto src_dst_tuple = cugraph::experimental::generate_rmat_edgelist<vertex_t>(
+    handle, scale, num_edges, a, b, c, seed, clip_and_flip, scramble_vertex_ids);
+
+  graph_generator_t gg_vals{
+    std::make_unique<rmm::device_buffer>(std::get<0>(src_dst_tuple).release()),
+    std::make_unique<rmm::device_buffer>(std::get<1>(src_dst_tuple).release())};
+
+  return std::make_unique<graph_generator_t>(std::move(gg_vals));
+}
+
+template <typename vertex_t>
+std::unique_ptr<graph_generator_t*> call_generate_rmat_edgelists(
```

Comment:
Why return pointer-to-pointer? This construct returns a `unique_ptr` to a raw pointer, `graph_generator_t*`. Why not just returning a `unique_ptr<graph_generator_t>` like at line 797? 

Also, you probably have a memory leak here, because for a `unique_ptr` to a pointer to get deallocated it needs an explicit `Deleter`, otherwise I don't see how that memory gets freed.

Reply:
The first one create a pointer to a struct containing `rmm::device_buffer` while the second one create a pointer to a pointer of struct containing `rmm::device_buffer`. The latter is pointing to a list of tuple (src,  dst) while the first is pointing to only one tuple (src, dst). There is definitely a leak and I am looking at how to incorporate some sort of unique pointer to my pointer which will automatically get deallocated at the end